includes/tracks/class-wc-site-tracking.php

```diff
@@ -21,8 +21,20 @@ public static function is_tracking_enabled() {
 		 * Don't track users who haven't opted-in to tracking or if a filter
 		 * has been applied to turn it off.
 		 */
-		if ( 'yes' !== get_option( 'woocommerce_allow_tracking' ) ||
-			! apply_filters( 'woocommerce_apply_user_tracking', true ) ) {
+
+		if (
+			! apply_filters( 'woocommerce_apply_user_tracking', true ) ||
+			(
+				'yes' !== get_option( 'woocommerce_allow_tracking' ) &&
+				// Check if tracking is actively being opted into.
+				// phpcs:disable WordPress.Security.NonceVerification.NoNonceVerification, WordPress.Security.ValidatedSanitizedInput
+				(
+					! isset( $_POST['wc_tracker_checkbox'] ) ||
+					'yes' !== sanitize_text_field( $_POST['wc_tracker_checkbox'] )
+				)
```

Comment:
Nice solution here! Its a bit hard to parse. How about something like this?

```php
if ( ! apply_filters( 'woocommerce_apply_user_tracking', true ) ) {
    return false;
}

// Check if tracking is actively being opted into.
$is_obw_opting_in = isset( $_POST['wc_tracker_checkbox'] ) && 'yes' === sanitize_text_field( $_POST['wc_tracker_checkbox'] );

if ( 'yes' !== get_option( 'woocommerce_allow_tracking' ) && ! $is_obw_opting_in ) {
   return false;
}
```

Reply:
Seems like a good way to handle this and much more readable - thanks! 
 Added in https://github.com/woocommerce/woocommerce/pull/22877/commits/848504222d76a65c09e8946ce30c3051a11c3cb0